extension/docs/time_weighted_average.md

```diff
@@ -0,0 +1,305 @@
+
+An aggregate that produces a `TimeWeightSummary` from timestamps and associated values. 
+
+### Required Arguments² [](time-weight-point-required-arguments)
+|Name| Type |Description|
+|---|---|---|
+| `method` | `TEXT` | The weighting method we should use, options are 'linear' or 'LOCF', not case sensitive |
+| `ts` | `TIMESTAMPTZ` |  The time at each point |
+| `value` | `DOUBLE PRECISION` | The value at each point to use for the time weighted average|
+<br>
+
+² Note that `ts` and `value` can be `null`, however the aggregate is not evaluated on `null` values and will return `null`, but it will not error on `null` inputs.
+
+### Returns
+
+|Column|Type|Description|
+|---|---|---|
+| `time_weight` | `TimeWeightSummary` | A TimeWeightSummary object that can be passed to other functions within the time weighting API. |
+<br>
+
+### Sample Usage
+```SQL
+WITH t as (
+    SELECT 
+        time_bucket('1 day'::interval, ts) as dt,   
+        timescale_analytics_experimental.time_weight('Linear', ts, val) AS tw -- get a time weight summary
+    FROM foo
+    WHERE id = 'bar'
+    GROUP BY time_bucket('1 day'::interval, ts)
+) 
+SELECT 
+    dt, 
+    timescale_analytics_experimental.average(tw) -- extract the average from the time weight summary
+FROM t;
+```
+
+## **time_weight() (summary form)** [](time-weight-summary)
+```SQL
+timescale_analytics_experimental.time_weight(
+    tws TimeWeightSummary
+) RETURNS TimeWeightSummary
+```
+
+An aggregate to compute a combined `TimeWeightSummary` from a series of non-overlapping `TimeWeightSummaries`. Non-disjoint `TimeWeightSummaries` will cause errors. See [Notes on Parallelism and Ordering](#time-weight-ordering) for more information.
+
+### Required Arguments² [](time-weight-summary-required-arguments)
+|Name| Type |Description|
+|---|---|---|
+| `tws` | `TimeWeightSummary` | The input TimeWeightSummary from a previous `time_weight` (point form) call, often from a [continuous aggregate](https://docs.timescale.com/latest/using-timescaledb/continuous-aggregates)|
+
+### Returns
+
+|Column|Type|Description|
+|---|---|---|
+| `time_weight` | `TimeWeightSummary` | A TimeWeightSummary object that can be passed to other functions within the time weighting API. |
+<br>
+
+### Sample Usage
+```SQL
+WITH t as (
+    SELECT 
+        date_trunc('day', ts) as dt,   
+        timescale_analytics_experimental.time_weight('Linear', ts, val) AS tw -- get a time weight summary
+    FROM foo
+    WHERE id = 'bar'
+    GROUP BY date_trunc('day')
+), q as (
+    SELECT timescale_analytics_experimental.time_weight(tw) AS full_tw -- do a second level of aggregation to get the full time weighted average
+    FROM t
+)
+SELECT 
+    dt, 
+    timescale_analytics_experimental.average(tw),  -- extract the average from the time weight summary
+    timescale_analytics_experimental.average(tw) / (SELECT full_tw FROM q LIMIT 1)  as normalized -- get the normalized average
+FROM t;
+```
```

Comment:
In general, I think this example would be clearer with the input out put pair; something like (note I have _not_ checked my math)
```SQL
WITH t as (
    SELECT 
        time_bucket('1 day'::interval, ts) as dt,   
        timescale_analytics_experimental.time_weight('Linear', ts, val) AS tw -- get a time weight summary
    FROM (VALUES
        ('2020-01-01 00:00',  0),
        ('2020-01-01 12:00', 20),
        ('2020-01-02 00:00', 5),
        ('2020-01-02 00:00', 10)
    ) points(ts, val)
    WHERE id = 'bar'
    GROUP BY time_bucket('1 day'::interval, ts)
) 
SELECT 
    dt, 
    timescale_analytics_experimental.average(tw) -- extract the average from the time weight summary
FROM t;
```
```output
 average
----------
      15
```